jvm/src/main/scala/io/glutenproject/expression/CoalesceOperatorTransformer.scala

```diff
@@ -38,7 +41,24 @@ class CoalesceTransformer(exps: Seq[Expression], original: Expression)
     with Logging {
 
   override def doTransform(args: java.lang.Object): ExpressionNode = {
-    throw new UnsupportedOperationException("Not supported: Coalesce.")
+    var notSupportFlag = false
+    val nodes = new java.util.ArrayList[ExpressionNode]()
+    val arrayBuffer = new ArrayBuffer[DataType]()
+    exps.foreach(expression => {
+      val expressionNode = expression.asInstanceOf[ExpressionTransformer].doTransform(args)
+      notSupportFlag = !expressionNode.isInstanceOf[ExpressionNode] || notSupportFlag
```

Comment:
Maybe just throw exception if not expression node?

Reply:
done